ios/insporationUITests/insporationUITests.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  insporationUITests.swift
+//  insporationUITests
+//
+//  Created by Thorsten Claus on 03.11.20.
+//
+
+import XCTest
+// Mark
+class insporationUITests: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+
+        // In UI tests it is usually best to stop immediately when a failure occurs.
+        continueAfterFailure = false
+
+        // In UI tests it’s important to set the initial state - such as interface orientation - required for your tests before they run. The setUp method is a good place to do this.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func testTakeSnapshots() throws {
+        // Assumtions:
+        // App is prepaired with an logged-in account, first tab (Main) is selected and Streams is selected
+  
+        let app = XCUIApplication()
+        app.launch()
+        setupSnapshot(app)
+        sleep(3) // Wait until stream is loaded
+        
+        snapshot("01FirstScreen")
+        
+        app.buttons["StreamType.main"].tap()
+        snapshot("02Selector")
+        app.otherElements["StreamType.followedTags"].tap()
+        sleep(2) // Wait until tags are loaded
+        snapshot("03Tags")
+        app.otherElements["Notifications\nNotifications\nTab 4 of 5"].tap()
```

Comment:
Maybe wouldn't hurt to use semantic labels here too? :)

Reply:
I found out that NavigationButtonItems in flutter doesn't have any semantics. It seems that visible text and keys is divided by \n . 
It turns out that after selecting a StreamType - button, it's not reliable to go back.

Reply:
Did you try to set the `semanticLabel` on the `Icon`? https://github.com/jhass/insporation/blob/e574c7eba6ae6435e435dcdedad1bd69fbbfef9e/lib/src/navigation.dart#L33

Reply:
Otoh it seems this is used by TalkOver and VoiceOver, so for blind users. Maybe we shouldn't misappropriate it for UI tests :/

Reply:
Does setting `key: Key("foo")` to stuff turn up in the widget tree visible to the UI tests?

Reply:
This is how flutter level integration tests find their widgets at least: https://flutter.dev/docs/cookbook/testing/integration/introduction